pkg/base1/test-import-json.ts

```diff
@@ -0,0 +1,227 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import { JsonValue } from 'cockpit';
+import {
+    import_json_object, import_optional, import_mandatory,
+    import_string, import_number, import_boolean,
+    import_array, import_record,
+    validate
+} from "import-json";
+
+import QUnit from 'qunit-tests';
+
+// Hook into console.error
+
+let console_errors: string[] = [];
+const console_error = console.error;
+console.error = function (msg) {
+    console_errors.push(msg);
+    console_error(msg);
+};
+
+QUnit.hooks.beforeEach(() => {
+    console_errors = [];
+});
+
+QUnit.test("import_string", function(assert) {
+    assert.equal(import_string("foo"), "foo", "string");
+    assert.throws(() => import_string(12), /JSON validation error for : Not a string: 12/, "not a string");
```

Comment:
I suggest to cover `""` as well, as that often triggers bugs with "falsy?" checks.